src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+		MPSNnScaleNode Create (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+
+		[Export ("initWithSource:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, MTLSize size);
+
+		[Export ("initWithSource:transformProvider:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNBinaryArithmeticNode")]
+	interface MPSNnBinaryArithmeticNode
+	{
+		[Static]
+		[Export ("nodeWithSources:")]
+		MPSNnBinaryArithmeticNode Create (MPSNnImageNode[] sourceNodes);
+
+		[Static]
+		[Export ("nodeWithLeftSource:rightSource:")]
+		MPSNnBinaryArithmeticNode Create (MPSNnImageNode left, MPSNnImageNode right);
+
+		[Export ("initWithSources:")]
+		IntPtr Constructor (MPSNnImageNode[] sourceNodes);
+
+		[Export ("initWithLeftSource:rightSource:")]
+		IntPtr Constructor (MPSNnImageNode left, MPSNnImageNode right);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNConcatenationNode")]
+	interface MPSNnConcatenationNode
+	{
+		[Static]
+		[Export ("nodeWithSources:")]
+		MPSNnConcatenationNode Create (MPSNnImageNode[] sourceNodes);
+
+		[Export ("initWithSources:")]
+		IntPtr Constructor (MPSNnImageNode[] sourceNodes);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNSoftMaxNode")]
+	interface MPSCnnSoftMaxNode
+	{
+		[Static]
+		[Export ("nodeWithSource:")]
+		MPSCnnSoftMaxNode Create (MPSNnImageNode sourceNode);
+
+		[Export ("initWithSource:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNLogSoftMaxNode")]
+	interface MPSCnnLogSoftMaxNode
+	{
+		[Static]
+		[Export ("nodeWithSource:")]
+		MPSCnnLogSoftMaxNode Create (MPSNnImageNode sourceNode);
+
+		[Export ("initWithSource:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNUpsamplingNearestNode")]
+	interface MPSCnnUpsamplingNearestNode
+	{
+		[Static]
+		[Export ("nodeWithSource:integerScaleFactorX:integerScaleFactorY:")]
+		MPSCnnUpsamplingNearestNode Create (MPSNnImageNode sourceNode, nuint integerScaleFactorX, nuint integerScaleFactorY);
+
+		[Export ("initWithSource:integerScaleFactorX:integerScaleFactorY:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint integerScaleFactorX, nuint integerScaleFactorY);
+
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNUpsamplingBilinearNode")]
```

Comment:
it looks like it's immutable so I would guess `init` (default constructor) does not do any good -> `[DisableDefaultCtor]`